kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -294,11 +306,17 @@ protected void writeAndFlushResponseToClient(Channel channel) {
                         response.getRequest().getHeader());
                 }
 
-                ByteBuf result = responseToByteBuf(response.getResponseFuture().get(), response.getRequest());
+                ByteBuf result = responseToByteBuf(response.getResponseFuture().get(
+                        kafkaConfig.getRequestTimeoutMs(),
+                        TimeUnit.MILLISECONDS), response.getRequest());
                 channel.writeAndFlush(result);
-            } catch (Exception e) {
-                // should not comes here.
-                log.error("error to get Response ByteBuf:", e);
+            } catch (ExecutionException | TimeoutException | InterruptedException e) {
+                // get response result may timeout.
+                log.error("request {}: error to get Response ByteBuf: ",
+                        response.getRequest().getHeader().apiKey(), e.getCause());
```

Comment:
```suggestion
                // get response result may timeout or fail
                if (e instanceof ExecutionException) {
                    log.error("request {}: error to get Response ByteBuf: ",
                            response.getRequest().getHeader(), e.getCause());
                } else {
                    log.error("request {} timed out or was interrupted", response.getRequest().getHeader());
                }
```

Two main changes:
1. Only log detail stacks for `ExecutionException`
2. Log the header instead of the api key, because the header contains correlation id that may be helpful to see which request failed.

Reply:
I got it, thanks for your suggestion.